### PR TITLE
feat: expose frame & move properties to console-message event object

### PR DESCRIPTION
**Pull request by samuelmaddock**
#### Description of Change

resolves #35418 

Implements what @MarshallOfSound proposed in #35418—this time moving properties to the event object.

The stack trace seemed [nuanced](https://source.chromium.org/chromium/chromium/src/+/main:content/public/browser/web_contents_observer.h;l=808-818;drc=8e948282d37c0e119e3102236878d6f4d5052c16) so I opted not to expose it.

> |untrusted_stack_trace| is not present for most messages; only when requested in advance and only for exceptions.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes:
* Added `frame` to 'console-message' event.
* Moved 'console-message' arguments into event object.

### CODE REVIEW

**Review by jkleinsc — CHANGES_REQUESTED** [review]
The deprecations need to be noted in sbreaking-changes.md

**Comment by samuelmaddock** [comment]
- updated to show deprecation warning when emitting to listeners with greater than one argument
- restored deprecated argument docs

cc @codebytere @itsananderson @erickzhao @jkleinsc

File: docs/api/web-contents.md [hunk 1]
```diff
@@ -949,11 +949,17 @@ Emitted when a `<webview>` has been attached to this web contents.
...
 
-* `event` Event
-* `level` Integer - The log level, from 0 to 3. In order it matches `verbose`, `info`, `warning` and `error`.
-* `message` string - The actual console message
-* `line` Integer - The line number of the source that triggered this console message
-* `sourceId` string
+* `details` Event\<\>
+  * `message` string - Message text
+  * `level` string - Message severity
+    Possible values include `info`, `warning`, `error`, and `debug`.
```
Comment (erickzhao) [inline docs/api/web-contents.md:955]: I think docs-parser will make this into a string enum if you format it similar to https://www.electronjs.org/docs/latest/api/app#appgetpathname

Comment (samuelmaddock) [reply docs/api/web-contents.md:955]: The possible values list gets parsed and turned into a string enum which is confirmed in the [Artifact Comparison check.](https://github.com/electron/electron/pull/43617/checks)

```
level: ('info' | 'warning' | 'error' | 'debug');
```

https://github.com/electron/typescript-definitions/blob/28f88c950a524689d920a8e26698367102c171bb/src/utils.ts#L148-L149

**Review by erickzhao — COMMENTED** [review]
API LGTM

File: docs/api/web-contents.md (hunk 1 above)
Comment (MarshallOfSound) [reply docs/api/web-contents.md:955]: Yeah the docs parser is actually quite loose here, if you wanna see the cursed regex that pulls this off --> https://github.com/electron/docs-parser/blob/0a1bd843256995399b01d55862736e359462b8ae/src/markdown-helpers.ts#L456-L478

**Review by jkleinsc — APPROVED** [review]
API LGTM

**Review by jkleinsc — APPROVED** [review]

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**